pytorch_lightning/callbacks/quantization.py

```diff
@@ -318,3 +326,82 @@ def on_predict_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule
+    storing tensors at lower bitwidths (such as INT8) than floating point precision. And this callback will
+    quantized model with Intel Neural Compressor(INC) tool.
+
+    Args:
+
+        config_path_or_obj: config file or Quantization_Conf class for INC:
+
+        metric_name: Specified metric which will be computed in evaluation function.
+
+        save_path: save path where the quantization config and weights be saved to
+
+    .. Intel Neural Compressor: https://github.com/intel/neural-compressor
+    """
+
+    def __init__(
+        self,
+        config_path_or_obj,
+        metric_name: str,
+        save_path: str = None,
+    ) -> None:
+        try:
+            from neural_compressor.conf.config import Quantization_Conf
+        except ImportError as e:
+            raise RuntimeError("Please install neural-compressor with: pip install neural-compressor") from e
+
+        self.config = (
+            config_path_or_obj
+            if isinstance(config_path_or_obj, Quantization_Conf)
+            else Quantization_Conf(config_path_or_obj)
+        )
+        self.approach = self.config.usr_cfg.quantization.approach
+        assert metric_name is not None, "Please set metric name for evaluation loop!"
+        self.metric_name = metric_name
+        self.save_path = save_path
+
+    def on_compress_model(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        """Called when the compressor loop begins."""
+        try:
+            from neural_compressor.experimental import common, Quantization
+        except ImportError as e:
+            raise RuntimeError("Please install neural-compressor with: pip install neural-compressor") from e
+
+        def eval_func(model):
+            pl_module_tmp = copy.deepcopy(pl_module)
+            pl_module_tmp.model = model
+            out = trainer.validate(pl_module_tmp)
+            trainer.state.fn = TrainerFn.COMPRESSING
+            return out[0][self.metric_name]
+
+        def train_func(model):
+            pl_module_tmp = copy.deepcopy(pl_module)
+            pl_module_tmp.model = model
+            trainer.fit(pl_module_tmp)
+            trainer.state.fn = TrainerFn.COMPRESSING
+            return pl_module_tmp.model
+
+        assert hasattr(
+            pl_module, "model"
+        ), "The LightningModule should have the backbone model which is a native PyTorch model!"
+        quantizer = Quantization(self.config)
+        quantizer.model = common.Model(pl_module.model)
+
+        if self.approach == QuantizationMode.PTQ_STATIC.value:
+            quantizer.calib_dataloader = pl_module.train_dataloader()
+
+        if self.approach == QuantizationMode.QAT.value:
+            quantizer.q_func = train_func
+        else:
+            trainer.post_quantizing = True
+
+        quantizer.eval_func = eval_func
+        model = quantizer()
+        pl_module.model = model.model
+        if self.save_path is not None:
+            model.save(self.save_path)
```

Comment:
This model is INC model and it will save quantized configure and weights. I think Trainer save_checkpoint can handle this case.

Reply:
I am afraid that Trainer save_checkpoint is not sufficient, since a complete quantized model for deployment needs both quantized model/weight and quantization config file. That’s why we introduce the save method for now, but we are open to use any official save API in the future and commit for the change, as long as it can support well for INT8 model deployment.